content/en/blog/_posts/2023-12-05-kubernetes-1.29.md

```diff
@@ -0,0 +1,183 @@
+	•	When a PersistentVolume represents encrypted block storage, for example using LUKS, you need to provide a passphrase in order to expand the device.
+	•	For various validations, the CSI driver needs to have credentials to communicate with the backend storage system at time of node expansion. 
+
+To meet this requirement, a CSI Node Expand Secret feature was introduced in Kubernetes 1.25. This allows an optional secret field to be sent as part of the NodeExpandVolumeRequest by the CSI drivers so that node volume expansion operation can be performed with the underlying storage system. In Kubernetes 1.29, this feature became generally available.
+
+## What's New in Beta
+
+### Per-plugin callback functions in kube-scheduler for accurate requeueing ([sig-scheduling](https://github.com/kubernetes/community/tree/master/sig-scheduling))
+
+The throughput of the scheduler is our eternal challenge. This QueueingHint feature brings a new possibility to optimize the efficiency of requeueing, which could reduce useless scheduling retries significantly.
+
+### Node lifecycle separated from taint management ([sig-scheduling](https://github.com/kubernetes/community/tree/master/sig-scheduling))
+
+As title describes, it's to decouple `TaintManager` that performs taint-based pod eviction from `NodeLifecycleController` and make them two separate controllers: `NodeLifecycleController` to add taints to unhealthy nodes and `TaintManager` to perform pod deletion on nodes tainted with NoExecute effect.
+
+## What's New in Alpha
+
+### Define Pod affinity or anti-affinity using `matchLabelKeys` ([sig-scheduling](https://github.com/kubernetes/community/tree/master/sig-scheduling)) 
+
+The throughput of the scheduler is our eternal challenge. This QueueingHint feature brings a new possibility to optimize the efficiency of requeueing, which could reduce useless scheduling retries significantly. We're actively working on this, and some stable in-tree plugins will have supported it in v1.29.
+
+### nftables backend for kube-proxy ([sig-network](https://github.com/kubernetes/community/tree/master/sig-network))
+    
+The default kube-proxy implementation on Linux is currently based on iptables. IPTables was the preferred packet filtering and processing system in the Linux kernel for many years (starting with the 2.4 kernel in 2001). However, unsolvable problems with iptables led to the development of a successor, nftables. Development on iptables has mostly stopped, with new features and performance improvements primarily going into nftables instead.
+
+This feature adds a new backend to kube-proxy based on nftables, since some Linux distributions already started to deprecate and remove iptables, and nftables claims to solve the main performance problems of iptables.
+    
+### An API to manage IP address ranges for Services ([sig-network](https://github.com/kubernetes/community/tree/master/sig-network))
```

Comment:
From the design and code implementation of [KEP-1880](https://github.com/kubernetes/enhancements/issues/1880), this feature introduces not one but two apis.
> The proposal is to implement a new allocator logic that uses 2 new API Objects: ServiceCIDR and IPAddress, and allow users to dynamically increase the number of Services IPs available by creating new ServiceCIDRs.

Reply:
Would this make more sense? Let me know, and I will commit the change. @mengjiao-liu @aojea 

`### Two APIs to manage IP address ranges for Services ([sig-network](https://github.com/kubernetes/community/tree/master/sig-network))

Services are an abstract way to expose an application running on a set of Pods. Services can have a cluster-scoped virtual IP address, that is allocated from a predefined CIDR defined in the `kube-apiserver` flags. However, users may want to add, remove or resize existing IP ranges allocated for Services without having to restart the `kube-apiserver`.

This feature implements a new allocator logic that uses 2 new API Objects: ServiceCIDR and IPAddress, and allows users to dynamically increase the number of Services IPs available by creating new ServiceCIDRs. This helps to resolve problems like IP exhaustion or IP renumbering.`

Reply:
It reads great to me, one clarification, the IPAddress object is not really user facing in this case (it may be user facing in the future), as the user can only interact with the ServiceCIDR objects

Reply:
So, an API client can't do a list or get on ipaddresses? (have I got that right?)

Reply:
oh, it can, I meant, there is no functionality associated to the object for the users,

Reply:
I've added a commit with the updated feature description, PTAL @aojea -- is any further clarification needed?